### ISSUE TITLE
Should FedoraMigrate handle PCDM migrations

### ISSUE THREAD

**Issue by awead** [projecthydra-labs/fedora-migrate#54]
This question was brought up in projecthydra-labs/curation_concerns#716

The scenarios are:
1. Migrate Sufia 5 to 6, or Fedora 3 to Fedora 4 (current)
2. Migrate Sufia 6 to 7, or Fedora 4 to PCDM
3. Migrate Sufia 5 to 7, or Fedora 3 to PCDM, aka. do not pass go, do not collect $100 dollars!

The original intention of the gem was just the first scenario. Whether or not it would hold up with the other two is unknown. We could build a new gem just for PCDM migrations, which would leave this gem only responsible for migrating Sufia from 5 to 6, then the new gem does 6 to 7.

Ping: @mjgiarlo @barmintor @hectorcorrea, who's currently working on some PCDM migration tools for Penn State.

**Comment by barmintor** [comment]
For my part, I think of this gem like ActiveFedora- mashing it up with Hydra::Works should probably be a downstream concern. That said, there are some things that will probably be required by PCDM migrations that make sense here.

**Comment by hectorcorrea** [comment]
I would vote for leaving this gem handling scenario 1 as it currently does and create separate gems for the other two scenarios.

Since the source data for scenarios 1 and 2 is totally different (Fedora 3 vs Fedora 4) I doubt there would be a significant amount of common functionality to justify a single gem, despite the fact that the output (Fedora 4 PCDM) would be the same. 

For the ScholarSphere migration from Sufia 6 to Sufia 7 I have been tinkering with exporting the`GenericFile` metadata to JSON files from a Sufia 6 instance and importing those files into Sufia 7 `GenericWork`, `FileSet`, and `File`. You can see here the proto-type for the *export* process here: https://github.com/psu-stewardship/scholarsphere/pull/272 and the one for the *import* process here https://github.com/projecthydra/sufia/blob/import_s6/lib/sufia/import/import_service.rb

**Comment by awead** [comment]
Yeah, I'm inclined to agree, @hectorcorrea. The other point is that most people will probably want to go from Sufia 5 to 7, and not stay on 6 forever. So I can see a migration path using a suite of tools such as this gem, with additional modifications as @barmintor points out, then another gem that does the PCDM lifting based on what you're working on.

**Comment by mjgiarlo** [comment]
:+1: to @hectorcorrea
